api/src/main/java/com/theokanning/openai/completion/chat/ChatCompletionChoice.java

```diff
@@ -25,4 +32,18 @@ public class ChatCompletionChoice {
      */
     @JsonProperty("finish_reason")
     String finishReason;
+
+    /**
+     * Custom response attributes that are not part of the core OpenAI API.
+     */
+    @Getter(onMethod = @__(@JsonAnyGetter))
```

Comment:
I just discovered that this causes problems for Javadoc. I'm hoping I can come up with a fix. If not, the alternative would be to add explicit fields for Azure OpenAI's extensions.

```
api/src/main/java/com/theokanning/openai/completion/chat/ChatCompletionChoice.java:39: error: cannot find symbol
    @Getter(onMethod = @__(@JsonAnyGetter))
                        ^
  symbol:   class __
  location: class ChatCompletionChoice
1 error
```

Reply:
I've fixed the Javadoc failure by adding a delombok step to build.gradle, which allows the Javadoc task to run against processed class files rather than the Lombok annotated originals.

This results in much more complete API Javadocs. Before:
<img width="1392" alt="before" src="https://user-images.githubusercontent.com/575245/234983425-426aba26-7eef-4427-b839-80fb93ab90e0.png">
After:
<img width="1392" alt="after" src="https://user-images.githubusercontent.com/575245/234983433-7c8cf5d0-18a2-4e74-a85f-55cbb1b478b8.png">

Alternatively, we could disable the Javadoc task entirely if you don't feel like this is a useful artifact to be publishing.